### DISCUSSION TITLE
DevOps not working correctly

### DISCUSSION THREAD

**Discussion by JanSzczepan** [Cyfrin/foundry-full-course-cu#751] [Q&A]
I import devops tools like this in my project:

```solidity
import {DevOpsTools} from "foundry-devops/src/DevOpsTools.sol";
```

and then:

```solidity
address contractAddress = DevOpsTools.get_most_recent_deployment(
         "BasicNft",
         block.chainid
);
```

I get this error:

```
├─ [22627] DevOpsTools::e374cdf1(00000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000007a69000000000000000000000000000000000000000000000000
000000000000000a426174746572794e667400000000000000000000000000000000000000000000) [delegatecall]
    │   ├─ [0] VM::ffi([pwd])
    │   │   └─ ← "Failed to execute command: program not found"
    │   └─ ← 0x0bc445030000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002c4661696c656420746f206578656375746520636f6d6d616e643a2070726f67
72616d206e6f7420666f756e640000000000000000000000000000000000000000
```

Any ideas?

**Comment by theirrationalone** [comment]
Please Add a new entry in your `foundry.toml` at the end:

```toml
ffi = true
```

and make sure, You have installed `jq`:

```bash
sudo apt-get install jq
```

**Reply by JanSzczepan** [reply]
Still the same error. It looks like a problem in DevOpsTools.sol in 79 line:
```solidity
string memory absolutePath = string(vm.ffi(pwd));
```
I think it can't find get_recent_deployment.sh script for some reason

**Reply by theirrationalone** [reply]
I think you have installed `devops` wrong version?? If so then, Please First delete `devops folder` from `lib folder` and then install this version:
```bash
forge install cyfrin/foundry-devops@0.0.11 --no-commit
```

**Reply by JanSzczepan** [reply]
Unfortunately still the same error. Any ideas?

**Reply by theirrationalone** [reply]
Heyy @JanSzczepan !, You're making NFTs right, Maybe you omitted to give access permissions....

In your `foundry.toml`
```foundry.tom
fs_permissions = [{ access = "read", path = "./images/" }] # path should your directory where your svgs/images resides.
```

But Traces don't tell that's the issue... I think You're right about `get_recent_deployment.sh script`.

Do you have this repo available?? If you like to share ???

**Reply by JanSzczepan** [reply]
Here is repo with the project:
https://github.com/JanSzczepan/devops-issue

Please help me fix this issue, because fs_permission is not a problem in my case

**Reply by theirrationalone** [reply]
Heyy @JanSzczepan !...

Sorry to say but you have highly messed up with your project.... it took appx 30+ mins... to setup again everything.... there are so many issues in your repo.... It would be better if you delete that project and start again from the beginning and please don't use `Prettier` here we're not dealing with `JS`, use solidity extension by `nomicfoundation` or use forge fmt to format your code.... or you can set rules for formatting your code with `forge fmt` into `foundry.toml` file.

And Please Don't feel Discouraged, I also have faced Similar kind of problem.

Please Make | Setup your Project again, You can copy your contracts btw. <3

**Reply by JanSzczepan** [reply]
Your answer is not helpful at all. Why my project is messed up? What should I do differently in setting up new one. Were you able to fix issue with devops?

**Reply by theirrationalone** [reply]
I just Copied your contracts and created a new project and that worked.... Your scripts were broken, I tried everything and exhausted then i made that new one.

**Reply by JanSzczepan** [reply]
Scripts work fine in my environment, interesting. Also I had to initialize my project using binary version of foundry downloaded from their GitHub, but it shouldn’t make any difference. Which version of foundry you use?

**Reply by theirrationalone** [reply]
version nightly, tag nightly-6c4c68a7031581bb8b8a10bb44db8dff4e04277 - my foundry version.

**Reply by theirrationalone** [reply]
Heyy @JanSzczepan ! What you did? what i did 🤣🤣, I went to check my foundry version, Eventually i made `foundryup` command to check the version into my bash. Now I have some other issues 😂, even my anvil, make commands,  aren't working, I'm facing `segmentation fault` error, Whenever i'm trying to run scripts explicitly on anvil, `make commands aren't reading `env variables` anymore. Bloody hell... new update is full of bugs.... 🐛

**Reply by JanSzczepan** [reply]
Yeah, that’s what I thought, I have to downgrade foundry and everything should work fine (I have the newest version currently and it’s buggy indeed)

**Reply by theirrationalone** [reply]
but i got holes in my rescue boat 😂😂.... downgrading too :)
